core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -147,6 +175,42 @@ static Snapshot fromJson(JsonNode node) {
     if (node.has(MANIFEST_LIST)) {
       // the manifest list is stored in a manifest list file
       String manifestList = JsonUtil.getString(MANIFEST_LIST, node);
+
+      ByteBuffer manifestListKeyMetadata = null;
+      ByteBuffer wrappedManifestListKeyMetadata = null;
+      String wrappedKeyEncryptionKey = null;
```

Comment:
Nit: Declare these inside `if (node.has(MANIFEST_LIST_KEY_METADATA))` ?

Reply:
They are used later, in the BaseSnapshot constructor. We can use a different constructor (if no manifest list encryption), but the code won't be more compact.